src/applications/personalization/profile360/actions/paymentInformation.js

```diff
@@ -0,0 +1,58 @@
+import { apiRequest } from 'platform/utilities/api';
+import { getData } from './index';
+
+export const FETCH_PAYMENT_INFORMATION_SUCCESS =
+  'FETCH_PAYMENT_INFORMATION_SUCCESS';
+
+export const SET_PAYMENT_INFO_UI_STATE = 'SET_PAYMENT_INFO_UI_STATE';
+
+export const SAVE_PAYMENT_INFORMATION = 'SAVE_PAYMENT_INFORMATION';
+export const SAVE_PAYMENT_INFORMATION_SUCCESS =
+  'SAVE_PAYMENT_INFORMATION_SUCCESS';
+export const SAVE_PAYMENT_INFORMATION_FAIL = 'SAVE_PAYMENT_INFORMATION_FAIL';
+
+export function fetchPaymentInformation() {
+  return async dispatch => {
+    dispatch({
+      type: FETCH_PAYMENT_INFORMATION_SUCCESS,
+      paymentInformation: await getData('/ppiu/payment_information'),
+    });
+  };
+}
+
+export function savePaymentInformation(fields) {
+  return async dispatch => {
+    const apiRequestOptions = {
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify(fields),
+      method: 'PUT',
+      mode: 'cors',
+    };
+
+    try {
+      dispatch({ type: SAVE_PAYMENT_INFORMATION });
+
+      const response = await apiRequest(
+        '/ppiu/payment_information',
+        apiRequestOptions,
+      );
+
+      dispatch({
+        type: SAVE_PAYMENT_INFORMATION_SUCCESS,
+        paymentInformation: response.data.attributes,
+      });
+    } catch (response) {
+      dispatch({
+        type: SAVE_PAYMENT_INFORMATION_FAIL,
+        response,
+      });
+    }
+  };
+}
+
+export function setPaymentInformationUiState(state) {
+  return {
+    type: SET_PAYMENT_INFO_UI_STATE,
+    state,
+  };
+}
```

Comment:
I wrote this action as sort of a replacement for `setState`, since the vets-website pattern is to keep most state in the store as opposed to component state.  It's used to pass data like the form fields, `isEditing, `isSaving`, etc.

Reply:
Our guidance is, generally, that action names should reflect something a user did or an external event. Meaning, a user didn't try to set the payment information ui state, they opened the edit view of the modal or saved something.

Reply:
Updated to use dedicated actions. This makes sense to me now that we're encouraging deriving state in the reducer, although I still think we can improve by setting guidelines for `setState`.